Improve security by using defusedxml when installed in environment

alorence opened the pull request:
This change was originally suggested by @atodorov, thanks to him for this contribution.

Currently, django-modern-rpc uses builtin `xmlrpc.client.loads()` function to parse incoming requests. But this can cause a security issue when an attacker send a valid but malicious XML-RPC request.

A Python package can help to mitigate this kind of attack, [defusedxml](https://pypi.org/project/defusedxml/) ([GitHub repo](https://github.com/tiran/defusedxml)).

The goal of this PR is to implement `defusedxml` support in django-modern-rpc

Currently:
- defusedxml is added to extra dependencies
- `defusedxml.xmlrpc.monkey_patch()` is called from `ModernRpcConfig.ready()` when available
- 2 functional tests have been added to ensure
  - use of XML `<!ENTITY key "value">` syntax in XML-RPC request is disabled and raise an exception on parsing step
  - the same payload can work when `defusedxml.xmlrpc.monkey_patch()` is not called, or when `defusedxml.xmlrpc.unmonkey_patch()` is called after project initialization

Remaining
- [x] Mention the change in the documentation
- [x] Improve comments and docstring, specifically around tests

Drawbacks
- ~~Due to a limitation in the way `live_server` fixture has been customized for these tests, Django 2.1 cannot be supported anymore~~ edit: this is not true anymore

atodorov commented on modernrpc/apps.py:
```diff
@@ -83,3 +84,13 @@ def import_modules(modules_list: List[str]) -> None:
+            logger.debug("'defusedxml' package were not found, no protection will be enabled against XML based attacks")
+            return
+        else:
+            defusedxml.xmlrpc.monkey_patch()
```
The `else` clause is redundant, see the comments from Codacy .

atodorov replied:
I think you can move this line immediately after line 91 where defusedxml is imported but that's more of a reference choice.

atodorov reviewed (commented):
Minor comments, otherwise LGTM.

alorence commented:
Thanks @atodorov for your feedbacks. I just implemented some of them, and renamed a few things. Some documentations and comments were also added.
In addition, I updated the way tests are run around this change, and I dropped the need to define a custom live_server fixture. So basically, we do not need to drop Django 2.1 support anymore...
